go/teams/teams.go

```diff
@@ -1569,8 +1486,117 @@ func isSigOldSeqnoError(err error) bool {
 	return false
 }
 
-func (t *Team) associateTLFID(ctx context.Context, tlfID keybase1.TLFID) (err error) {
-	defer t.G().CTrace(ctx, "Team.associateTLFID", func() error { return err })()
+func (t *Team) marshal(incoming interface{}) ([]byte, error) {
+	var data []byte
+	mh := codec.MsgpackHandle{WriteExt: true}
+	enc := codec.NewEncoderBytes(&data, &mh)
+	if err := enc.Encode(incoming); err != nil {
+		return nil, err
+	}
+	return data, nil
+}
+
+func (t *Team) boxKBFSCryptKeys(ctx context.Context, key keybase1.TeamApplicationKey,
+	kbfsKeys []keybase1.CryptKey) (string, string, error) {
+
+	marshaledKeys, err := t.marshal(kbfsKeys)
+	if err != nil {
+		return "", "", err
+	}
+
+	var nonce [libkb.NaclDHNonceSize]byte
+	if _, err := rand.Read(nonce[:]); err != nil {
+		return "", "", err
+	}
+
+	var encKey [libkb.NaclSecretBoxKeySize]byte = key.Material()
+	sealed := secretbox.Seal(nil, marshaledKeys, &nonce, &encKey)
+	dat := keybase1.TeamEncryptedKBFSKeyset{
+		V: 1,
+		N: nonce[:],
+		E: sealed,
+	}
+
+	marshaledSealedDat, err := t.marshal(dat)
+	if err != nil {
+		return "", "", err
+	}
+
+	encStr := base64.StdEncoding.EncodeToString(marshaledSealedDat)
+	sbytes := sha256.Sum256([]byte(encStr))
+	hashStr := hex.EncodeToString(sbytes[:])
```

Comment:
are we doing hex or base64 here?  I thought base64?

Reply:
This is the hash, the encryted_keyset itself is base64.

Reply:
Guess I can b64 it too.

Reply:
Nevermind, server wants hex.